pkg/platform/runtime/runtime.go

```diff
@@ -173,8 +173,12 @@ func (r *Runtime) recordCompletion(err error) {
 		return
 	}
 	r.completed = true
+
 	logging.Debug("Recording runtime completion: %v", err == nil)
 
+	// Record attempt regarless of success
+	r.recordAttempt()
```

Comment:
Sorry for not flagging this before; but this should fire at the start of the "attempt". ie. the second the runtime is starting to be used. Now it is firing after the actual attempt has happened, the only distinction is that it doesn't trigger if the attempt was unsuccessful. 

The intent of the event is to signal that an attempt IS being made. Not that an attempt WAS made. This is important because any number of things could have happened in between.

Reply:
No problem, I think the `recordAttempt` calls are in the right place now.